.buildkite/pipelines/sonarqube.yml.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+
+cat <<EOL
+steps:
+  - label: "Generate compile_commands.json"
+    key: "export_compile_commands"
+    depends_on: "check_style"
+    agents:
+      image: "docker.elastic.co/ml-dev/ml-linux-build:29"
+    command:
+      - "cmake CMAKE_EXPORT_COMPILE_COMMANDS=ON -B cmake-build-docker"
+    artifact_paths:
+      - "cmake-build-docker/compile_commands.json"
+    notify:
+      - github_commit_status:
+          context: "Generate compile_commands.json"
+  - label: "Static code analysis with SonarQube :sonarqube:"
+    key: "sonarqube"
+    depends_on: "export_compile_commands"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "secret/ci/elastic-ml-cpp/sonar-analyze-token"
+    command: 
+      - "buildkite-agent artifact download cmake-build-docker/compile_commands.json ."
+      - ".buildkite/scripts/steps/sonar-qube.sh"
```

Comment:
I _think_ this script has been missed from the PR?

Reply:
Indeed. It should be "/scan-source-code.sh" which is available in the sonar docker image.